packages/app/src/store/run-all-specs-store.ts

```diff
@@ -0,0 +1,96 @@
+import { RUN_ALL_SPECS_KEY } from '@packages/types/src'
+import { gql, useMutation, useQuery } from '@urql/vue'
+import { defineStore } from 'pinia'
+import { computed, ref } from 'vue'
+import { useRouter } from 'vue-router'
+import { RunAllSpecsDataDocument, RunAllSpecsDocument } from '../generated/graphql-test'
+import { getSeparator, SpecTreeNode, UseCollapsibleTreeNode } from '../specs/tree/useCollapsibleTree'
+
+type ResolvedConfig = { value: any, from: 'string', field: string }[]
+
+gql`
+query RunAllSpecsData {
+  currentProject {
+    id
+    config
+    currentTestingType
+  }
+}
+`
+
+gql`
+mutation RunAllSpecs ($specPath: String!, $runAllSpecs: [String!]!) {
+  setRunAllSpecs(runAllSpecs: $runAllSpecs)
+  launchOpenProject(specPath: $specPath) {
+    id
+  }
+}
+`
+
+const isRunMode = window.__CYPRESS_MODE__ === 'run' && window.top === window
+
+export const useRunAllSpecsStore = defineStore('runAllSpecs', () => {
+  const allSpecsRef = ref<string[]>([])
+  const directoryChildrenRef = ref<Record<string, string[]>>({})
+
+  const separator = getSeparator()
+  const router = useRouter()
+  const query = useQuery({ query: RunAllSpecsDataDocument, pause: isRunMode })
+  const setRunAllSpecsMutation = useMutation(RunAllSpecsDocument)
+
+  async function runSpecs (runAllSpecs: string[]) {
+    await setRunAllSpecsMutation.executeMutation({ runAllSpecs, specPath: RUN_ALL_SPECS_KEY })
+
+    // Won't execute unless we are testing since the browser gets killed. In testing,
+    // we can stub `launchProject` to verify the functionality is working
+    router.push({ path: '/specs/runner', query: { file: RUN_ALL_SPECS_KEY } })
+  }
+
+  async function runAllSpecs () {
+    await runSpecs(allSpecsRef.value)
+  }
+
+  async function runSelectedSpecs (dir: string) {
+    await runSpecs(directoryChildrenRef.value[dir])
+  }
+
+  function setRunAllSpecsData (tree: UseCollapsibleTreeNode<SpecTreeNode>[]) {
+    const allSpecs: string[] = []
+    const directoryChildren: Record<string, string[]> = {}
+
+    for (const { id, isLeaf } of tree) {
+      if (!isLeaf) {
+        directoryChildren[id] = []
+      } else {
+        allSpecs.push(id)
+
+        Object.keys(directoryChildren).forEach((dir) => {
+          if (id.startsWith(dir) && id.replace(dir, '').startsWith(separator)) {
+            directoryChildren[dir].push(id)
+          }
+        })
+      }
+    }
+
+    allSpecsRef.value = allSpecs
+    directoryChildrenRef.value = directoryChildren
+  }
+
+  const isRunAllSpecsAllowed = computed(() => {
+    const isE2E = query.data.value?.currentProject?.currentTestingType === 'e2e'
+
+    const config: ResolvedConfig = query.data.value?.currentProject?.config || []
+    const hasExperiment = config.find(({ field, value }) => field === 'experimentalRunAllSpecs' && value === true)
+
+    return Boolean(isE2E && hasExperiment)
+  })
+
+  return {
+    isRunAllSpecsAllowed,
+    allSpecs: allSpecsRef,
```

Comment:
Seems `allSpecs` is not used outside the store; a private ref - should we not return it here (thus not exposing it)?

Reply:
You're right! It became unused after a refactor